src/packagedef.jl

```diff
@@ -262,10 +267,13 @@ function delete_missing!(exs_sigs_old::ExprsSigs, exs_sigs_new)
                         @debug "DeleteMethod" _group="Action" time=time() deltainfo=(sig, MethodSummary(m))
                         # Delete the corresponding methods
                         for p in workers()
-                            try  # guard against serialization errors if the type isn't defined on the worker
-                                remotecall(Core.eval, p, Main, :(delete_method_by_sig($sig)))
-                            catch
-                            end
+                            #try  # guard against serialization errors if the type isn't defined on the worker
+                                future = remotecall(Core.eval, p, Main, :(delete_method_by_sig($sig)))
+                                finalizer(future) do f
+                                    Base.invoke_in_world(worldage[], Distributed.finalize_ref, f)
+                                end
+                            #catch
```

Comment:
Do we need to catch all errors here, or can we only catch certain ones? This bit me during debugging, when I tried to throw an error in the finalizer to get a stacktrace for debugging.

Reply:
Certainly it's fine to comment out while debugging, or to `catch e; isa(e, SomeKindOfError) && throw(e)` if `SomeKindOfError` deserves special handling. What kind of errors did you have in mind?

Off-topic, but since we talked about `backtrace()` in our other discussion, just checking, do you know about `catch_backtrace()`? It's sometimes quite useful.

Reply:
I was more wondering, what kind of errors this was initially supposed to catch, because catching all possible errors seems seems a bit dangerous to me, since we possibly ignore errors the user would like to know about. So I was more thinking of something along the lines of `e isa ErrorThatsDistributedsFault || rethrow()`, i.e. that the default would be to rethrow errors. Perhaps that's just not possible in this case though.

> Off-topic, but since we talked about backtrace() in our other discussion, just checking, do you know about catch_backtrace()?

I do now! :laughing:

Reply:
Or can we at least log those errors with something like `@warn`?

Reply:
I see your point. This probably is the "lazy" approach, but doing better requires that we anticipate two things:

- it seems likely that many errors might have already been caught by the main process, and so a report from a worker would be redundant. However, I suspect this was more of a concern in previous versions of Revise where there was more `try/catch` stuff; now, it seems likely that it wouldn't even get here if it fails in the main process.
- workers may not all have the same code---for example, in some cases I've deliberately loaded Gtk only in the main process and had the workers be just computational engines. In such cases, some revise attempts will fail, but it would be wrong to send a warning to the user who would mentally think "I know, I did it on purpose."

As long as we can avoid pestering users incorrectly, I'm all in favor of pestering them when they deserve it :smile:.

Reply:
Is there a way to check whether an error was thrown during regular "revising" or something else? If not, I would propose we just keep it as it is currently, but use either `@debug` or Revise's logging facilities to log the error, so people who want to know about these kinds of errors can still get them.

Reply:
The pkgdata might be in `revision_errors` but the needed info doesn't propagate down here. I think logging it makes sense.